executor/containerdexecutor/executor.go

```diff
@@ -329,5 +295,63 @@ func (w *containerdExecutor) Exec(ctx context.Context, id string, process execut
 	if err != nil {
 		return errors.WithStack(err)
 	}
-	return taskProcess.Start(ctx)
+
+	err = w.runProcess(ctx, taskProcess, process.Resize, nil)
+	return err
+}
+
+func (w *containerdExecutor) runProcess(ctx context.Context, p containerd.Process, resize <-chan executor.WinSize, started func()) error {
+	statusCh, err := p.Wait(context.Background())
+	if err != nil {
+		return err
+	}
+
+	err = p.Start(ctx)
+	if err != nil {
+		return err
+	}
+
+	if started != nil {
+		started()
+	}
+
+	p.CloseIO(ctx, containerd.WithStdinCloser)
+
+	var cancel func()
+	ctxDone := ctx.Done()
+	for {
+		select {
+		case <-ctxDone:
+			ctxDone = nil
+			var killCtx context.Context
+			killCtx, cancel = context.WithTimeout(context.Background(), 10*time.Second)
+			p.Kill(killCtx, syscall.SIGKILL)
+		case size := <-resize:
+			err := p.Resize(ctx, size.Cols, size.Rows)
+			if err != nil {
+				cancel()
+				return err
+			}
```

Comment:
I think there might be a race where a client-size window resize happens right around the same time the task exits, in which case the `resize` chan could get triggered first and then `p.Resize` returns an expected error because the task process has exited. That would result in the resize error getting returned instead of the actual task status (which could have been an expected exit with 0 status).

I think it also wouldn't hurt to have a timeout on the `Resize` call just in case it hangs for unexpected reasons.

Reply:
Yeah, good catch.  I wonder if we should just ignore resize errors in general, maybe just log them?  If Resize is not working the user will likely be having other issues anyway.  I will add a `context.WithTimeout`, should probably also throw it in a goroutine to prevent it blocking the cancel/exit handling.

Reply:
I updated the resize to run later in select loop and to run in short lived goroutine with 1s timeout.